File: DecompiledGameScripts/cyberpunk/devices/masters/systems/securitySystem/securitySystem.swift

public class SecuritySystem extends DeviceSystemBase {

  private let m_savedOutputCache: [OutputValidationDataStruct];

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as SecuritySystemController;
  }

  public const func GetDefaultHighlight() -> ref<FocusForcedHighlightData> {
    return null;
  }

  public const func GetDevicePS() -> ref<SecuritySystemControllerPS> {
    return this.GetController().GetPS();
  }

  private const func GetController() -> ref<SecuritySystemController> {
    return this.m_controller as SecuritySystemController;
  }

  protected cb func OnSlaveStateChanged(evt: ref<PSDeviceChangedEvent>) -> Bool {
    return false;
  }

  protected cb func OnQuestIllegalActionNotification(evt: ref<QuestIllegalActionNotification>) -> Bool {
    this.GetDevicePS().QuestIllegalActionNotification(evt);
  }

  protected cb func OnQuestCombatActionNotification(evt: ref<QuestCombatActionNotification>) -> Bool {
    this.GetDevicePS().QuestCombatActionNotification(evt);
  }

  protected cb func OnSetSecuritySystemState(evt: ref<SetSecuritySystemState>) -> Bool {
    this.GetDevicePS().QuestChangeSecuritySystemState(evt);
  }

  protected cb func OnQuestAuthorizePlayer(evt: ref<AuthorizePlayerInSecuritySystem>) -> Bool {
    this.GetDevicePS().QuestAuthorizePlayer(evt);
  }

  protected cb func OnQuestBlackListPlayer(evt: ref<BlacklistPlayer>) -> Bool {
    this.GetDevicePS().QuestBlacklistPlayer(evt);
  }

  protected cb func OnQuestExclusiveQuestControl(evt: ref<SuppressSecuritySystemStateChange>) -> Bool {
    this.GetDevicePS().QuestSuppressSecuritySystem(evt);
  }

  protected cb func OnQuestChangeSecuritySystemAttitudeGroup(evt: ref<QuestChangeSecuritySystemAttitudeGroup>) -> Bool {
    this.GetDevicePS().QuestChangeSecuritySystemAttitudeGroup(evt);
  }

  public func OnMaraudersMapDeviceDebug(sink: ref<MaraudersMapDevicesSink>) -> Void;
}

File: Definitions/Game_Definitions/classes/SecuritySystem.lua
---@meta
---@diagnostic disable

---@class SecuritySystem : DeviceSystemBase
---@field savedOutputCache OutputValidationDataStruct[]
SecuritySystem = {}

---@return SecuritySystem
function SecuritySystem.new() return end

---@param props table
---@return SecuritySystem
function SecuritySystem.new(props) return end

---@param evt AuthorizePlayerInSecuritySystem
---@return Bool
function SecuritySystem:OnQuestAuthorizePlayer(evt) return end

---@param evt BlacklistPlayer
---@return Bool
function SecuritySystem:OnQuestBlackListPlayer(evt) return end

---@param evt QuestChangeSecuritySystemAttitudeGroup
---@return Bool
function SecuritySystem:OnQuestChangeSecuritySystemAttitudeGroup(evt) return end

---@param evt QuestCombatActionNotification
---@return Bool
function SecuritySystem:OnQuestCombatActionNotification(evt) return end

---@param evt SuppressSecuritySystemStateChange
---@return Bool
function SecuritySystem:OnQuestExclusiveQuestControl(evt) return end

---@param evt QuestIllegalActionNotification
---@return Bool
function SecuritySystem:OnQuestIllegalActionNotification(evt) return end

---@param evt SetSecuritySystemState
---@return Bool
function SecuritySystem:OnSetSecuritySystemState(evt) return end

---@param evt gamePSDeviceChangedEvent
---@return Bool
function SecuritySystem:OnSlaveStateChanged(evt) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function SecuritySystem:OnTakeControl(ri) return end

---@return SecuritySystemController
function SecuritySystem:GetController() return end

---@return FocusForcedHighlightData
function SecuritySystem:GetDefaultHighlight() return end

---@return SecuritySystemControllerPS
function SecuritySystem:GetDevicePS() return end

---@param sink worldMaraudersMapDevicesSink
function SecuritySystem:OnMaraudersMapDeviceDebug(sink) return end

